src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx

```diff
@@ -15,43 +24,189 @@ const ContactTagChip = styled(Chip)(({ theme }) => ({
   marginLeft: '0',
 }));
 
-const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
+const ContactTagIcon = styled(TagIcon)(({ theme }) => ({
+  color: theme.palette.cruGrayMedium.main,
+  marginRight: theme.spacing(1),
+}));
+
+const ContactTagInput = styled(TextField)(({ theme }) => ({
+  '&& .MuiFormLabel-root.Mui-focused': {
+    color: theme.palette.cruGrayDark.main,
+  },
+  '&& .MuiInput-underline:before ': {
+    borderBottom: 'none',
+  },
+  '&& .MuiInput-underline:after ': {
+    borderBottom: `2px solid ${theme.palette.divider}`,
+  },
+  '&& .MuiInputBase-input': {
+    marginBottom: '13px',
+  },
   margin: theme.spacing(1),
   marginLeft: '0',
 }));
 
 interface ContactTagsProps {
+  accountListId: string;
   contactId: string;
   contactTags: string[];
 }
 
+const tagSchema = yup.object({
+  tag: yup.string().required(),
+});
+
 export const ContactTags: React.FC<ContactTagsProps> = ({
+  accountListId,
   contactId,
   contactTags,
 }) => {
   const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const [
+    updateContactTags,
+    { loading: updating },
+  ] = useUpdateContactTagsMutation();
+
   const handleTagDelete = (tag: string) => {
     const index = contactTags.indexOf(tag);
+
     if (index > -1) {
-      contactTags = contactTags.splice(index, 1);
-      // TODO: add mutation to update tag for contact
+      const tagList = [...contactTags];
+      tagList.splice(index, 1);
+      try {
+        updateContactTags({
+          variables: {
+            accountListId,
+            contactId,
+            tagList,
+          },
+          update: (cache, { data: updateData }) => {
```

Comment:
I think this whole update function is unnecessary. Apollo seems to handle updating the query fine on it's own. Probably because this isn't a relationship to another type in the cache and is just an array of strings.

You can `await` the update and have the snackbar call after.

Reply:
Ah, interesting. So unless I'm updating some other query that may contain data that was altered, I don't need to manually update the cache.